### ISSUE TITLE
Wrong aws_transfer_user user_name validation

### ISSUE THREAD

**Issue by pjanuario** [hashicorp/terraform-provider-aws#16917]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform CLI and Terraform AWS Provider Version

```
Terraform v0.13.1
+ provider registry.terraform.io/hashicorp/aws v3.19.0
```

### Affected Resource(s)

* aws_transfer_user

### Terraform Configuration Files

```hcl
resource "aws_transfer_user" "user" {
  server_id            = "s-0123456789abcdef0"
  user_name         = "user_name"
  role                     = "arn:aws:iam::123456789012:role/transfer"
  home_directory = "/bucket/path"
}
```

### Expected Behavior

Terraform creates the user

```
 + aws_transfer_user.user
      id:             <computed>
      arn:            <computed>
      home_directory: "/bucket/path"
      role:           "arn:aws:iam::123456789012:role/transfer"
      server_id:      "s-0123456789abcdef0"
      user_name:      "user_name"
```

### Actual Behavior

Terraform fails to create users containing between 32 and 100 characters

```
Error: Invalid "user_name": must be between 3 and 32 alphanumeric or special characters hyphen and underscore. However, "user_name" cannot begin with a hyphen
```

### Steps to Reproduce

1. `terraform apply`

### References
AWS updated their user_name validation: https://docs.aws.amazon.com/transfer/latest/userguide/API_CreateUser.html#API_CreateUser_RequestSyntax

UserName
A unique string that identifies a user and is associated with a as specified by the ServerId. This user name must be a minimum of 3 and a maximum of 100 characters long. The following are valid characters: a-z, A-Z, 0-9, underscore '_', hyphen '-', period '.', and at sign '@'. The user name can't start with a hyphen, period, or at sign.

Type: String

Length Constraints: Minimum length of 3. Maximum length of 100.

Pattern: ^[\w][\w@.-]{2,99}$

Required: Yes

**Comment by pjanuario** [comment]
It looks like we just need to update the ValidateFunc for the username attribute. 

 https://github.com/hashicorp/terraform-provider-aws/blob/master/aws/validators.go#L97

**Comment by bflad** [comment]
The update for this has been merged and will release with version 3.23.0 of the Terraform AWS Provider, later this week. Thank you to @pjanuario for the implementation. 👍
